src/strategies/strategy-univ3-base.sol

```diff
@@ -0,0 +1,413 @@
+        require(msg.sender == governance || msg.sender == strategist || harvesters[msg.sender], "not authorized");
+
+        for (uint256 i = 0; i < _harvesters.length; i++) {
+            harvesters[_harvesters[i]] = true;
+        }
+    }
+
+    function revokeHarvesters(address[] calldata _harvesters) external {
+        require(msg.sender == governance || msg.sender == strategist, "not authorized");
+
+        for (uint256 i = 0; i < _harvesters.length; i++) {
+            harvesters[_harvesters[i]] = false;
+        }
+    }
+
+    function setWithdrawalDevFundFee(uint256 _withdrawalDevFundFee) external {
+        require(msg.sender == timelock, "!timelock");
+        withdrawalDevFundFee = _withdrawalDevFundFee;
+    }
+
+    function setWithdrawalTreasuryFee(uint256 _withdrawalTreasuryFee) external {
+        require(msg.sender == timelock, "!timelock");
+        withdrawalTreasuryFee = _withdrawalTreasuryFee;
+    }
+
+    function setPerformanceDevFee(uint256 _performanceDevFee) external {
+        require(msg.sender == timelock, "!timelock");
+        performanceDevFee = _performanceDevFee;
+    }
+
+    function setPerformanceTreasuryFee(uint256 _performanceTreasuryFee) external {
+        require(msg.sender == timelock, "!timelock");
+        performanceTreasuryFee = _performanceTreasuryFee;
+    }
+
+    function setStrategist(address _strategist) external {
+        require(msg.sender == governance, "!governance");
+        strategist = _strategist;
+    }
+
+    function setGovernance(address _governance) external {
+        require(msg.sender == governance, "!governance");
+        governance = _governance;
+    }
+
+    function setNFTFee(uint256 _fee) external {}
+
+    function setTimelock(address _timelock) external {
+        require(msg.sender == timelock, "!timelock");
+        timelock = _timelock;
+    }
+
+    function setController(address _controller) external {
+        require(msg.sender == timelock, "!timelock");
+        controller = _controller;
+    }
+
+    function getProportion() public view returns (uint256) {
+        (uint256 a1, uint256 a2) = pool.amountsForLiquidity(1000000000000000000, tick_lower, tick_upper);
+        return (a2 * (10**18)) / a1;
+    }
+
+    function _wrapAllToNFT() internal returns (uint256, uint256) {
+        (uint256 _tokenId, uint256 _liquidity) = _wrapSomeToNFT(
+            token0.balanceOf(address(this)),
+            token1.balanceOf(address(this))
+        );
+        return (_tokenId, _liquidity);
+    }
+
+    function _wrapSomeToNFT(uint256 _amount0, uint256 _amount1) internal returns (uint256, uint256) {
+        token0.safeApprove(address(nftManager), 0);
+        token0.safeApprove(address(nftManager), _amount0);
+
+        token1.safeApprove(address(nftManager), 0);
+        token1.safeApprove(address(nftManager), _amount1);
+
+        (uint256 _tokenId, uint128 _liquidity, , ) = nftManager.mint(
```

Comment:
If the strategy has already created an initial liquidity NFT position, should we call `increaseLiquidity` when subsequently adding liquidity?